vaadin-grid-flow-parent/vaadin-grid-flow/src/main/resources/META-INF/resources/frontend/gridConnector.js

```diff
@@ -731,8 +731,24 @@ import { ItemCache } from '@vaadin/vaadin-grid/src/vaadin-grid-data-provider-mix
           }
           updateGridItemsInDomBasedOnCache(items);
         }
+        const cacheAndIndex = grid._cache.getCacheAndIndexByKey(pkey);
+        clearSubcache(cacheAndIndex.cache, cacheAndIndex.scaledIndex, index, length);
       });
 
+      /*
+       * Deletes items from grid cache, including the descendants of the affected
+       * items.
+       */
+      function clearSubcache(cache,scaledIndex, index, length) {
+        const end = length + index;
+        const childCache = cache.itemCaches[scaledIndex];
+        for(let itemIndex = index; itemIndex < end; itemIndex++) {
+          delete childCache.items[itemIndex];
+          delete childCache.itemCaches[itemIndex];
```

Comment:
The connector seems to add a property called `itemkeyCaches` to all caches, which is a map of key-cache pairs. Should that also be covered in the clearing logic (it would include references to caches that are deleted here)?

Reply:
Done